File: frontend/src/services/UserService.ts
import CommonServiceBase from "./CommonServiceBase";
import UserData from "@/models/UserData";

export default class UserService extends CommonServiceBase {
  public constructor() {
    super("/user");
  }

  public async getUserData(): Promise<UserData> {
    return this.get<undefined, UserData>("");
  }
}

File: backend/Backend/Services/UserService.cs
public class UserService : IUserService {
  private IQuizService quizService;
  private IRoomService roomService;
  private IGameService gameService;

  public UserService(IQuizService quizService, IRoomService roomService, IGameService gameService) {
    this.quizService = quizService;
    this.roomService = roomService;
    this.gameService = gameService;
  }

  public void InitUser(string user) {
    lock (this) {
      quizService.InitUser(user);
      roomService.InitUser(user);
      gameService.InitUser(user);
    }
  }

  public UserData GetUserData(string user) {
    lock (this) {
      return new UserData {
        Username = user,
        Quizzes = quizService.GetQuizzesInfo(user)
      };
    }
  }
}
